Userland/Libraries/LibTest/Randomized/Chunk.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2023, Martin Janiczek <martin@janiczek.cz>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Format.h>
+#include <AK/String.h>
+
+namespace Test {
+namespace Randomized {
+
+// Chunk is a description of a RandomRun slice.
+// Used to say which part of a given RandomRun will be shrunk by some
+// ShrinkCommand.
+//
+// For a RandomRun [0,1,2,3,4,5,6,7,8], the Chunk{size=4, index=2} means this:
+//                 [_,_,X,X,X,X,_,_,_]
+//
+// Different ShrinkCommands will use the Chunk in different ways.
+// A few examples:
+//
+//     Original RandomRun:             [5,1,3,9,4,2,3,0]
+//     Chunk we'll show off:           [_,_,X,X,X,X,_,_]
+//
+//     ZeroChunk:                      [5,1,0,0,0,0,3,0]
+//     SortChunk:                      [5,1,2,3,4,9,3,0]
+//     DeleteChunkAndMaybeDecPrevious: [5,1,        3,0]
+struct Chunk {
+    uint8_t size;
+    size_t index;
```

Comment:
(Dang, a reply I've sent didn't get through. Let's write this again.)

Chunk sizes currently only are 1,2,3,4,8. We could have larger sizes, it brings larger risk (of the RandomRun not being parsable into a value) but also larger reward (more skipped/fast-forwarded shrinking work).

But instead of doing that, a better approach is to copy what Python Hypothesis does, and record start,end indexes of each generator call, and make chunks from _those_. Haven't done that here, but it would be a nice next optimization (not only speed wise, but also shrink quality wise).